### DISCUSSION TITLE
Filter out outbound message based on specific ship type

### DISCUSSION THREAD

**Discussion by jherichan** [jvde-github/AIS-catcher#221] [Ideas]
Due to certain local restrictions, looking for a configurable parameter than can block the message from specific vessel types being sent in outbound (-S or -u), in order to comply to local restrictions while still be able to share received messages to others.
Messages from all types of vessels can be processed and stored locally, only all the message from specified types of vessels cannot be forwarded/shared to others.
Otherwise the station would have to be dropped from community. Thanks!

**Comment by jvde-github** [comment]
Thanks for raising. Would like to understand the precise requirement a bit better. How do we recognize messages that cannot be forwarded precisely? MMSI and msg type are quite straightforward as they are contained in every message.

You mention "shiptype", this is not included in all messages. So the idea would then be to not forward data until we have received a message that a particular ship is not of the sensitive type. This is more tricky. So, the question really is what is the easiest way to identify these vessels to implement somehow.

**Reply by jherichan** [reply]
It was induced from the consideration of the recent regulatory activities:
https://www.scmp.com/news/china/politics/article/3247525/china-targets-devices-it-says-are-used-send-flight-data-foreign-entities
Without completely dropped out from community, so looking for a filtering mechanism to keep the station 'alive'.

While considering specific MMSI of the concern subject cannot be known in advance, and they will broadcast ordinary messages as other type of vessels, for safe navigation purpose. So I look for ways other than MMSI or MSG TYPE filter.

Thanks for your quick response and considering the idea.

**Reply by Taxom** [reply]
I have a question for them, how are they going to limit receiving of AIS messages by satellites? ))))
![ais-sat](https://github.com/jvde-github/AIS-catcher/assets/20313019/808fcd5e-ff3b-43ca-b6c8-dade92b352eb)

**Reply by jherichan** [reply]
I would read the concern are on the data being collected locally, and ultimate receiving party.

**Reply by jherichan** [reply]
Have a second thought on my wish idea, I think I should rather implement a middleware to bridge the outbound data and external community, instead of a feature natively in the ais-catcher app. The original idea can be dropped.

(I like this app as it greatly improved AIS data volume.)
